Fix broken image link and markdown

jzold opened the pull request:
This PR covers issue #192 .

- [x] This PR handles the issue and requires no additional PRs.
- [x] You have validated the need for this change.

**What did this PR accomplish?**

- Fix markdown issues in 4.12.13_Testing_for_Cross_Site_Script_Inclusion_OTG-CLIENT-013.md
- Fix broken link in 4.12.13_Testing_for_Cross_Site_Script_Inclusion_OTG-CLIENT-013.md

Thank you for your contribution!

jzold commented:
please only review/approve (as appropriate)  2099910 only - cherry picking is required :)

kingthorin commented on document/4_Web_Application_Security_Testing/4.12_Client_Side_Testing/4.12.13_Testing_for_Cross_Site_Script_Inclusion_OTG-CLIENT-013.md:
```diff
@@ -2,7 +2,7 @@
 Cross Site Script Inclusion (XSSI) vulnerability allows sensitive data leakage across-origin or cross-domain boundaries. Sensitive data could include authentication related data (login states, cookies, auth tokens, session IDs, etc.), user personal or sensitive personal data (email addresses, phone numbers, credit card details, social security numbers, etc.). XSSI is a client-side attack similar  ...
 
-By default websites are only allowed to access data if they are from the same origin. This is a key application security principle and governed by the same-origin policy (defined by RFC 6454 - [https://tools.ietf.org/html/rfc6454]). An origin is defined as the combination of URI scheme (http/https), host name, and port number. However, this policy is bypassed/not applicable for <script> HTML tag  ...
+By default websites are only allowed to access data if they are from the same origin. This is a key application security principle and governed by the same-origin policy (defined by RFC 6454 - <https://tools.ietf.org/html/rfc6454>). An origin is defined as the combination of URI scheme (http/https), host name, and port number. However, this policy is bypassed/not applicable for <script> HTML tag  ...
```
Please use a properly formatted markdown link.

As well as backticks on the script tag.

jzold replied:
@kingthorin  re markdown link: previously the link was defined with [ ]. I'm using vscode with markdownlint to reduce the number of markdown issues I had on my PRs so far (and thanks @kingthorin @ThunderSon for helping and pointing those out :)  markdownlint recommends <> for links (see https://github.com/DavidAnson/markdownlint/blob/v0.17.0/doc/Rules.md#md034), should I be sticking with [ ] as it was previously? Is there a markdown/format guide for OWASP?

kingthorin replied:
So 034 is right it shouldn't be a bare link. I'm sure &lt; &gt; is syntactically correct, however, I believe the preference is something like: `[defined by RFC 6454](https://tools.ietf.org/html/rfc6454)`

There's nothing published yet but the template PR does cover this: https://github.com/OWASP/OWASP-Testing-Guide-v5/pull/102/

(Yes we appreciate and acknowledge that you can't implement or necessarily be aware of something that isn't published.)

jzold replied:
@kingthorin thanks, I'm reverting the URL back to [ ]

jzold replied:
My linter is showing an error - links with  [ ] are marked as bare links by markdownlint :) For now I'll set an exception for this markdown error

kingthorin replied:
That's why I provided an example you could use :+1: 
`[defined by RFC 6454](https://tools.ietf.org/html/rfc6454)`

jzold replied:
@kingthorin thanks again, it seems I still have a long way to go with markdown :)
Slightly off topic, I saw in the past OAuth vulnerabilities was on the OWASP todos list and I don't know if there was any action/is there any work undertaken by anyone at this moment, I don't have visibility, please advise :)

kingthorin replied:
There's a TODO item on the Project Summit 2017 details to update the auth testing section and add OAUTH details. If you want to create an issue to track against and tackle the changes or authoring then go for it. 

https://github.com/OWASP/OWASP-Testing-Guide-v5/blob/master/OWASP_Summit_Outcomes.md#outcome

kingthorin commented on document/4_Web_Application_Security_Testing/4.12_Client_Side_Testing/4.12.13_Testing_for_Cross_Site_Script_Inclusion_OTG-CLIENT-013.md:
```diff
@@ -135,9 +139,9 @@ Gmail had a similar vulnerability in 2006 that allowed the extraction of user co
 ## Example Leak Sensitive Data via Javascript Runtime Errors
 
-Browsers normally present standardized Javascript error messages like "Script error.". However, in the case of IE9/10 runtime error messages provided additional details which could be used to leak data. For example, a website (victim.com) serves the following content at the URI http://victim.com/service/csvendpoint for authenticated users:
+Browsers normally present standardized Javascript error messages like "Script error.". However, in the case of IE9/10 runtime error messages provided additional details which could be used to leak data. For example, a website (victim.com) serves the following content at the URI <http://victim.com/service/csvendpoint> for authenticated users:
```
Please use a properly formatted markdown link.

jzold replied:
see previous comment/question

kingthorin replied:
This one could be code fenced with backticks, since it's an example it doesn't need to be clickable.

jzold replied:
fixed in  5205ad7

kingthorin commented on document/4_Web_Application_Security_Testing/4.12_Client_Side_Testing/4.12.13_Testing_for_Cross_Site_Script_Inclusion_OTG-CLIENT-013.md:
The `./` can be dropped.

kingthorin replied:
Ex: https://github.com/OWASP/OWASP-Testing-Guide-v5/pull/102#discussion_r329052185

kingthorin commented:
In the future you should create a feature branch before starting your modifications.

kingthorin commented:
Looks good to me, if you can just clean up the commits.

jzold commented:
@kingthorin do you mean removing/rebasing the old commits on the PR?

kingthorin commented:
Ya so that the PR contains only actual new changes.

kingthorin commented:
I think I can fix it if you need.

jzold commented:
@kingthorin I rebased it and deleted all the commits in between to start with a "clean slate".  It might not have been the easiest approach but hopefully the latest commit contains all fixes without the old commits :)

kingthorin commented:
We still have one merge there that seems off but I don't mind fixing that if you'd like. If you're happy with it, 8eb88b6 looks right to me.

jzold commented:
@kingthorin this must be the longest PR in the markdown OWASP history. Please remove the first merge 0926d (much appreaciated) and in the future I'll make sure old commits don't stick around :)

kingthorin commented:
No problem. Trust me I’ve been through some nightmare PRs in this project and zaproxy too. Never fear we’re a community and hit harder together!

kingthorin commented:
For next time, things will be easier if you make sure that your local master is up-to-date then create a feature branch to work on.

kingthorin commented:
Fixed. If there's further feedback from others I don't mind tackling it. Otherwise make sure you do a fetch from origin and reset your local branch based on it before doing anything or else it clobber the tweak.

jzold commented:
thanks @kingthorin for all your help, "together we hit harder" I couldn't have said it better myself...though this is enough more than enough hitting for one day :)

jzold commented:
@kingthorin - my latest commit shows dismissed on your review I hope this does not cause any issues 
@ThunderSon  - thanks for the feedback, amended

kingthorin commented:
That's totally fine, we'ce set the repo to dismiss reviews when new contributions are added. All good.
